core/domain/improvements_services.py

```diff
@@ -212,3 +213,27 @@ def apply_changes_to_model(task_entry, task_entry_model):
         task_entry_model.resolved_on = task_entry.resolved_on
         changes_were_made_to_model = True
     return changes_were_made_to_model
+
+
+def update_task_dict_with_resolver_settings(task_entry):
+    """Makes changes to the given task dict according to resolver settings.
```

Comment:
This is not really what it does, It returns a task entry dict where username and profile pictures is inserted. Also I think this can be moved to _controllers/improvements.py_